src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/AlwaysGenerateDocumentsService.java

```diff
@@ -41,25 +42,26 @@ public AlwaysGenerateDocumentsService(
     }
 
     @Override
-    public byte[] generateClaimIssueReceipt(String externalId, String authorisation) {
-        return claimIssueReceiptService.createPdf(getClaimByExternalId(externalId, authorisation));
+    public byte[] generateDocument(String externalId, ClaimDocumentType claimDocumentType, String authorisation) {
+        switch (claimDocumentType) {
+            case CLAIM_ISSUE_RECEIPT:
+                return claimIssueReceiptService.createPdf(
+                    getClaimByExternalId(externalId, authorisation)).getBytes();
```

Comment:
getClaimByExternalId(externalId, authorisation) is common, so can be moved as single statement before switch

Reply:
👍